File: 2022-11-21-spring-session-with-redis/action-in-blog/src/main/java/action/in/blog/config/WebMvcConfig.java
package action.in.blog.config;

import action.in.blog.filter.SessionFilter;
import org.springframework.boot.web.servlet.FilterRegistrationBean;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class WebMvcConfig {

    @Bean
    public FilterRegistrationBean setFilterRegistration() {
        FilterRegistrationBean filterRegistrationBean = new FilterRegistrationBean(new SessionFilter());
        filterRegistrationBean.addUrlPatterns("/*");
        return filterRegistrationBean;
    }
}

File: 2024-07-16-how-to-deploy-react-application-on-jar-package/action-in-blog/backend-maven/src/main/kotlin/action/in/blog/config/WebMvcConfig.kt
package action.`in`.blog.backend.config

import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Profile
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Profile(value = ["pcf"])
@Configuration
class WebMvcConfig : WebMvcConfigurer {

    override fun addCorsMappings(registry: CorsRegistry) {
        registry.addMapping("/**")
            .allowedOrigins(
                "https://todos-frontend.apps.sandbox.aws.maki.lol",
                "https://todos-frontend.apps.dhaka.cf-app.com"
            )
            .allowedMethods("GET", "POST", "OPTIONS")
    }
}